dashboard/alerts/alert.py

```diff
@@ -87,29 +119,41 @@ def saveAlert(self):
         """
         try:
             # Open config file to get path
-            config_file = open('/etc/capstone-ids/config.json', 'r')
-            config_data = json.load(config_file)
-            path = config_data["alert_collection_path"]
-
-            # Default path
-            if path == "" or path is None:
-                path = "/etc/capstone-ids/alert_collection.json"
+            try:
+                config_file = open('/etc/tinyHIPPO/config.json', 'r')
+                config_data = json.load(config_file)
+                path = config_data["alert_collection_path"]
+            except:
+                path = "/etc/tinyHIPPO/alert_collection.json"
 
             # Open alert collection file to read
-            alert_collection = open(path, 'r')
-            alert_data = json.load(alert_collection)
+            alert_collection = None
+            try:
+                alert_collection = open(path, 'r')
+                try:
+                    alert_data = json.load(alert_collection)
+                except json.decoder.JSONDecodeError:
+                    alert_data = json.loads("{\n \"alerts\": []\n}\n")
+            except FileNotFoundError:
+                alert_data = json.loads("{\n \"alerts\": []\n}\n")
```

Comment:
Should the path that "alert_collect" needs to open ever not exist? IE is this something that needs to be initialized on setup of our script? If so that might make this block of code unneeded since you'll be able to initialize the required fields.

Reply:
*alert_collection

Reply:
The file would exist on installation, but that wouldn't stop the user from being able to delete it which would cause an error crashing the program. The user could also change the alert_path, since it is a configurable value, to something that doesn't exist which would cause an error. So rather than a crash it'd be easier to handle the error.

Reply:
Where would the user change alert_path? If it's in the config then our script would need to be reinitialized regardless I believe. Also IMO if the user deletes the file at that path then they would need to restart the program, seems like an error that shouldn't be caught considering it's where we're storing every alert in this case right?

Reply:
The file needs to be present regardless, and since this should be the only access point for the file I don't see why we need to limit the creation of the file to just initialization. And if the user deletes the file at that path, what would restarting the program gain us? The file and its data would still be gone and the program would just be needlessly restarted. 

All of this btw is completely temporary considering we plan to switch to the SQL based system at some point in the future, so I don't see much to gain by setting up a file initialization script just to remove it later. When using the SQL service I 1200% agree that the burden of starting and ensuring operation should be during initialization, but that's a whole nother development task.